### ISSUE TITLE
"npm test" failing

### ISSUE THREAD

**Issue by ghatwala** [strongloop/strong-pm#289]
Hi All,

I am getting following errors when i run "npm test " for this repo.
I have installed/started docker daemon on a red hat vm ,  
$ npm test
5c70d8b212205e52.1446026400475
2015-10-28T10:00:04.467Z pid:8374 worker:1 ENV.supervisor_profile: undefined
2015-10-28T10:00:04.467Z pid:8374 worker:1 ENV.PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/root/strongloop/strong-pm/node_mbin:/usr/local/mysql/bin:/usr/local/rvm/gems/ruby-2.1.2/bin:/usr/local/rvm/gems/ruby-2.1.2@global/bin:/usr/local/rvm/rubies/ruby-2.1.2//local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/usr/local/rvm/bin:/root/bin
2015-10-28T10:00:04.467Z pid:8374 worker:1 ENV.PWD: /root/strongloop/strong-pm/d20151028-8110-9msks50/.strong-pm/svc/1/work/current
2015-10-28T10:00:04.471Z pid:8374 worker:1 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-10-28T10:00:04.471Z pid:8374 worker:1 Please contact sales@strongloop.com for assistance.
2015-10-28T10:00:04.482Z pid:8374 worker:1 pid 8374 listening on 3001
sl-pm.js: Service "1" listening on *:3001

./test/test-pmctl-ssh2.js ............................. 0/1
  not ok timeout

Bail out! # timeout
Any pointers to resolve the above issue will be helpful ?

**Comment by sam-github** [comment]
Try it on a mac or linux first before mixing docker into the problem.

**Comment by rmg** [comment]
The ssh test assumes that the current user can ssh in to localhost as itself using private/public key authentication.

**Comment by rmg** [comment]
For the ssh tests, the underlying module is strong-tunnel. There's an outstanding wish list item to add server side support (just backlogged it as strongloop/strong-tunnel#4) which would allow the assumptions the pm tests make to be lifted.

**Comment by ghatwala** [comment]
@sam-github : my set up is NAME="Red Hat Enterprise Linux Server"
VERSION="7.1 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.1"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
@rmg  : i was under the impression that this was test case was failing as i hadn't installed docker on my vm . if you just guide me how resolve the above test case would be helpful.

**Comment by rmg** [comment]
The tests that use docker are in `./test/test-e2e-docker.sh` but the failure you are seeing is in `./test/test-pmctl-ssh2.js'.

The ssh2 tests require that your current user be able to run `ssh localhost` without a password.

If this is not possible, then it would probably be best to skip that test (either by marking it as skipped or simply deleting the file).

**Comment by ghatwala** [comment]
Hi @rmg ,

Thanks for the reply !!!
 i was able to SSH to localhost using below
generate the public private keys
> ssh-keygen -t dsa -P '' -f ~/.ssh/id_dsa
authorize the key by adding it to the list of authorized keys
> cat ~/.ssh/id_dsa.pub >> ~/.ssh/authorized_keys
test that you can log in with no password
> ssh localhost
Check to make sure this works (doesn’t prompt for password)

After doing above , i ran "npm test" again - it appears to me "ssh2" tests are passing however there seems other failures , can u please comment on below ?

/*************************/

./test/test-pmctl-ssh2.js working dir for /root/strongloop/strong-pm/test/test-pmctl-ssh2.js is /root/strongloop/strong-pm
exec `git clean -f -d -x .`
exec `git init`
exec `git add .`
exec `git commit --author="sl-pm-test <nobody@strongloop.com>" -m initial`
exec `node "/root/strongloop/strong-pm/node_modules/strong-build/bin/sl-build.js" --install --commit`
test/app built succesfully
pmcli: /root/strongloop/strong-pm/bin/sl-pm.js [ '--base=.strong-pm', '--listen=0' ]
sl-pm.js(9640): StrongLoop PM v5.0.1 (API v6.1.0) on port `59555`
sl-pm.js(9640): Base folder `/root/strongloop/strong-pm/d20151029-9472-10r1q3i0/.strong-pm`
sl-pm.js(9640): Applications on port `3000 + service ID`
Browse your REST API at http://127.0.0.1:59555/explorer
Listening port: 59555
pmurl: http://127.0.0.1:59555
Run request for commit "1/deploy/default/master" on current (none)
Start Runner: commit 1/deploy/default/master
2015-10-29T09:51:57.403Z pid:9707 worker:0 INFO strong-agent v2.0.2 profiling app 'test-app' pid '9707'
2015-10-29T09:51:57.430Z pid:9707 worker:0 INFO strong-agent[9707] started profiling agent
2015-10-29T09:51:57.431Z pid:9707 worker:0 INFO supervisor starting (pid 9707)
2015-10-29T09:51:57.432Z pid:9707 worker:0 INFO strong-agent strong-agent using strong-cluster-control v2.2.1
2015-10-29T09:51:57.435Z pid:9707 worker:0 INFO supervisor reporting metrics to `internal:`
2015-10-29T09:51:57.443Z pid:9707 worker:0 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-10-29T09:51:57.443Z pid:9707 worker:0 Please contact sales@strongloop.com for assistance.
2015-10-29T09:51:57.475Z pid:9707 worker:0 INFO supervisor size set to 1
2015-10-29T09:51:57.704Z pid:9707 worker:0 INFO supervisor started worker 1 (pid 9712)
2015-10-29T09:51:57.705Z pid:9707 worker:0 INFO supervisor resized to 1
2015-10-29T09:51:57.964Z pid:9712 worker:1 INFO strong-agent v2.0.2 profiling app 'test-app' pid '9712'
2015-10-29T09:51:57.964Z pid:9712 worker:1 INFO strong-agent[9712] started profiling agent
2015-10-29T09:51:57.978Z pid:9712 worker:1 PID 9712
2015-10-29T09:51:57.978Z pid:9712 worker:1 CWD /root/strongloop/strong-pm/d20151029-9472-10r1q3i0/.strong-pm/svc/1/work/ca327f3abb3516bff5c58e2e871a294137a941.1446112314529
2015-10-29T09:51:57.978Z pid:9712 worker:1 ENV.supervisor_profile: undefined
2015-10-29T09:51:57.979Z pid:9712 worker:1 ENV.PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/root/strongloop/strong-pm/nodeodules/.bin:/usr/local/mysql/bin:/usr/local/rvm/gems/ruby-2.1.2/bin:/usr/local/rvm/gems/ruby-2.1.2@global/bin:/usr/local/rvm/rubies/ry-2.1.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/usr/local/rvm/bin:/root/bin
2015-10-29T09:51:57.979Z pid:9712 worker:1 ENV.PWD: /root/strongloop/strong-pm/d20151029-9472-10r1q3i0/.strong-pm/svc/1/work/current
2015-10-29T09:51:57.983Z pid:9712 worker:1 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-10-29T09:51:57.983Z pid:9712 worker:1 Please contact sales@strongloop.com for assistance.
2015-10-29T09:51:57.994Z pid:9712 worker:1 pid 9712 listening on 3001
sl-pm.js: Service "1" listening on *:3001
./test/test-pmctl-ssh2.js ............................. 0/1
  not ok timeout

Bail out! # timeout
npm ERR! Test failed.  See above for more details.


/*************************/

**Comment by rmg** [comment]
It would probably be best just to skip this one test then.

Also keep in mind that strong-pm isn't meant to be run as root and therefore also isn't meant to be tested as root.

**Comment by ghatwala** [comment]
Hi @rmg ,

I have rerun the above test using normal user and skipping the above test case , but am getting some licensing errors , can you please comment on the below ?

/***********************************************/

./test/test-server-metadata.js ...................... 27/28
  objects-start 1.1.1
  not ok Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:51775 objects-start 1.1.1 Command "objects-start" on "http://127.0.0.1:51775" failed with Error: Object tracking requires license
    found:
      stack: >-
        Error: Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:51775
        objects-start 1.1.1
        Command "objects-start" on "http://127.0.0.1:51775" failed with Error: Object
        tracking requires license
            at ChildProcess.exithandler (child_process.js:203:12)
           at emitTwo (events.js:87:13)
           at ChildProcess.emit (events.js:172:7)
           at maybeClose (internal/child_process.js:817:16)
           at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
      message: >
        Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:51775
        objects-start 1.1.1
        Command "objects-start" on "http://127.0.0.1:51775" failed with Error: Object
        tracking requires license
      killed: false
      code: 1
      signal: null
      cmd: '/bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:51775 objects-start 1.1.1'
    at:
      file: /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js
      line: 25
      column: 9
    source: |
      t.ifError(err);
    stack: |
      /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js:25:9
      ChildProcess.exithandler (child_process.js:210:5)
      maybeClose (internal/child_process.js:817:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

Bail out! # Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:51775 objects-start 1.1.1 Command "objects-start" on "http://127.0.0.1:51775" failed with Error: Object tracking requires license
npm ERR! Test failed.  See above for more details.
/************************************************/

Any pointers on above ?

**Comment by ghatwala** [comment]
Hi @rmg  ,

Could you please help in pointing me to the proper path as  how to acquire "object tracking license" and ensure that the test cases  of "strong-pm " are passing locally ?

**Comment by rmg** [comment]
The license warnings aren't fatal - there is no license required to run tests.

To see what is happening, try running the failing test directly and with debug logging enabled:
```
DEBUG=strong-pm:test node test/test-pmctl-ssh2.js
```

**Comment by ghatwala** [comment]
Hi @rmg,

I ran the above command , please see the error encountered below:

/*************************/
[test@pts00449-vm19 strong-pm]$ DEBUG=strong-pm:test node test/test-pmctl-ssh2.js
TAP version 13
    # Subtest: pmctl over ssh
working dir for /home/test/strongloop_pck/strong-pm/test/test-pmctl-ssh2.js is /home/test/strongloop_pck/strong-pm
exec `git clean -f -d -x .`
exec `git init`
exec `git add .`
exec `git commit --author="sl-pm-test <nobody@strongloop.com>" -m initial`
exec `node "/home/test/strongloop_pck/strong-pm/node_modules/strong-build/bin/sl-build.js" --install --commit`
test/app built succesfully
pmcli: /home/test/strongloop_pck/strong-pm/bin/sl-pm.js [ '--base=.strong-pm', '--listen=0' ]
sl-pm.js(8334): StrongLoop PM v5.0.1 (API v6.1.0) on port `58725`
sl-pm.js(8334): Base folder `/tmp/d20151104-8168-1e7drteg/.strong-pm`
sl-pm.js(8334): Applications on port `3000 + service ID`
Browse your REST API at http://127.0.0.1:58725/explorer
Listening port: 58725
pmurl: http://127.0.0.1:58725
# running queued tests... (1)
# START waiton cmd "status 1" pattern /1\.1\.\d+\s+\d+\s+1\s+ \d+\.\d+\.\d+\.\d+:\d+/g
Run request for commit "1/deploy/default/master" on current (none)
Start Runner: commit 1/deploy/default/master
  strong-pm:test Run: --control,http+ssh://127.0.0.1:58725,status,1 => 0 out <
>
err <
ssh connection error:  { [Error: All configured authentication methods failed] level: 'client-authentication' }
> +0ms
2015-11-04T08:36:19.601Z pid:8401 worker:0 INFO strong-agent v2.0.2 profiling app 'test-app' pid '8401'
2015-11-04T08:36:19.612Z pid:8401 worker:0 INFO strong-agent[8401] started profiling agent
2015-11-04T08:36:19.613Z pid:8401 worker:0 INFO supervisor starting (pid 8401)
2015-11-04T08:36:19.615Z pid:8401 worker:0 INFO strong-agent strong-agent using strong-cluster-control v2.2.1
2015-11-04T08:36:19.617Z pid:8401 worker:0 INFO supervisor reporting metrics to `internal:`
2015-11-04T08:36:19.626Z pid:8401 worker:0 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-11-04T08:36:19.626Z pid:8401 worker:0 Please contact sales@strongloop.com for assistance.
2015-11-04T08:36:19.628Z pid:8401 worker:0 INFO supervisor size set to 1
2015-11-04T08:36:19.835Z pid:8401 worker:0 INFO supervisor started worker 1 (pid 8406)
2015-11-04T08:36:19.836Z pid:8401 worker:0 INFO supervisor resized to 1
2015-11-04T08:36:20.079Z pid:8406 worker:1 INFO strong-agent v2.0.2 profiling app 'test-app' pid '8406'
2015-11-04T08:36:20.094Z pid:8406 worker:1 INFO strong-agent[8406] started profiling agent
2015-11-04T08:36:20.105Z pid:8406 worker:1 PID 8406
2015-11-04T08:36:20.105Z pid:8406 worker:1 CWD /tmp/d20151104-8168-1e7drteg/.strong-pm/svc/1/work/347aa771e72db20a89f54fa0a6af21ddd35cbfb6.1446626176671
2015-11-04T08:36:20.106Z pid:8406 worker:1 ENV.supervisor_profile: undefined
2015-11-04T08:36:20.106Z pid:8406 worker:1 ENV.PATH: /usr/local/rvm/gems/ruby-2.1.2/bin:/usr/local/rvm/gems/ruby-2.1.2@global/bin:/usr/local/rvm/rubies/ruby-2.1.2/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/local/rvm/bin:/home/test/.local/bin:/home/test/bin
2015-11-04T08:36:20.106Z pid:8406 worker:1 ENV.PWD: /tmp/d20151104-8168-1e7drteg/.strong-pm/svc/1/work/current
2015-11-04T08:36:20.110Z pid:8406 worker:1 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-11-04T08:36:20.110Z pid:8406 worker:1 Please contact sales@strongloop.com for assistance.
2015-11-04T08:36:20.119Z pid:8406 worker:1 pid 8406 listening on 3001
sl-pm.js: Service "1" listening on *:3001
  strong-pm:test Run: --control,http+ssh://127.0.0.1:58725,status,1 => 0 out <
>
err <
ssh connection error:  { [Error: All configured authentication methods failed] level: 'client-authentication' }
> +3s
  strong-pm:test Run: --control,http+ssh://127.0.0.1:58725,status,1 => 0 out <
>
err <
ssh connection error:  { [Error: All configured authentication methods failed] level: 'client-authentication' }
> +2s
^C
/*****************************/

Any pointers to resolve the above error ?

**Comment by rmg** [comment]
@ghatwala try doing `ssh-add ~/.ssh/id_rsa` to add your private key to your ssh-agent, assuming you have one running.

If you are unfamiliar with ssh agents, this page seems to have a decent overview: https://developer.github.com/guides/using-ssh-agent-forwarding/

**Comment by ghatwala** [comment]
@rmg : Thanks for the above , the test case  "test-pmctl-ssh2.js"
[test@pts00449-vm19 strong-pm]$ DEBUG=strong-pm:test node test/test-pmctl-ssh2.js
Passes with no errors , after i started ssh agent "exec ssh-agent bash" and later adding private as suggested above .

However , the over npm test is still failing with below -
/**********************************************************************/
./test/test-server-metadata.js ...................... 27/28
  objects-start 1.1.1
  not ok Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:60935 objects-start 1.1.1 Command "objects-start" on "http://127.0.0.1:60935" failed with Error: Object tracking requires license
    found:
      stack: >-
        Error: Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:60935
        objects-start 1.1.1
        Command "objects-start" on "http://127.0.0.1:60935" failed with Error: Object
        tracking requires license
        at ChildProcess.exithandler (child_process.js:203:12)
        at emitTwo (events.js:87:13)
        at ChildProcess.emit (events.js:172:7)
        at maybeClose (internal/child_process.js:817:16)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
      message: >
        Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:60935
        objects-start 1.1.1
        Command "objects-start" on "http://127.0.0.1:60935" failed with Error: Object
        tracking requires license
      killed: false
      code: 1
      signal: null
      cmd: '/bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:60935 objects-start 1.1.1'
    at:
      file: /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js
      line: 25
      column: 9
    source: |
      t.ifError(err);
    stack: |
      /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js:25:9
      ChildProcess.exithandler (child_process.js:210:5)
      maybeClose (internal/child_process.js:817:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

Bail out! # Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:60935 objects-start 1.1.1 Command "objects-start" on "http://127.0.0.1:60935" failed with Error: Object tracking requires license
npm ERR! Test failed.  See above for more details.
/**********************************************************************/
Next when i debugged the "test/test-server-metadata.js" , using below , found errors as below
[test@pts00449-vm19 strong-pm]$ DEBUG=strong-pm:test node test/test-server-metadata.js
we have 2 failed test cases out of 27 tests , namely giving one failure-

/**********************************************************************/
Run: /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313 objects-start 1.1.1 => err: {"killed":false,"code":1,"signal":null,"cmd":"/bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313 objects-start 1.1.1"} stdout <
>    not ok 1 - Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313 objects-start 1.1.1 Command "objects-start" on "http://127.0.0.1:50313" failed with Error: Object tracking requires license
      ---
      found:
        stack: >-
          Error: Command failed: /bin/sh -c
          /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313
          objects-start 1.1.1
          Command "objects-start" on "http://127.0.0.1:50313" failed with Error: Object
          tracking requires license
              at ChildProcess.exithandler (child_process.js:203:12)
              at emitTwo (events.js:87:13)
              at ChildProcess.emit (events.js:172:7)
              at maybeClose (internal/child_process.js:817:16)
              at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
        message: >
          Command failed: /bin/sh -c
          /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313
          objects-start 1.1.1
          Command "objects-start" on "http://127.0.0.1:50313" failed with Error: Object
          tracking requires license
        killed: false
        code: 1
        signal: null
        cmd: '/bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:50313 objects-start 1.1.1'
      at:
        file: /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js
        line: 25
        column: 9
      source: |
        t.ifError(err);
      stack: |
        /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js:25:9
        ChildProcess.exithandler (child_process.js:210:5)
        maybeClose (internal/child_process.js:817:16)
        Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
      ...
    1..1
    # failed 1 of 1 tests
not ok 12 - objects-start 1.1.1 # time=1065.706ms
/***********************************************************/
My set up is -
[test@pts00449-vm19 strong-pm]$ lscpu
Architecture:          ppc64le
Byte Order:            Little Endian
CPU(s):                1
On-line CPU(s) list:   0
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             1
NUMA node(s):          1
Model:                 IBM pSeries (emulated by qemu)
L1d cache:             64K
L1i cache:             32K
NUMA node0 CPU(s):     0

It appears that error - " object tracking requires license " , could you please let me know how can we resolve this error ?

**Comment by rmg** [comment]
@ghatwala I've opened a PR to skip tests that require licenses: #291. Can you try checking out that branch (`licenseless-tests`) and seeing if it passes for you?

Additionally, you seem to be testing this on POWER, which has not been tested by StrongLoop or IBM yet.

**Comment by ghatwala** [comment]
@rmg , thanks for the reply.
[test@pts00449-vm19 strong-pm]$ git branch
* licenseless-tests
  master

Next i ran the same npm test command , however i still get errors.
/**************************/
[test@pts00449-vm19 strong-pm]$ npm test
>
./test/test-server-metadata.js ...................... 27/28
  objects-start 1.1.1
  not ok Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:53628 objects-start 1.1.1 Command objects-start" on "http://127.0.0.1:53628" failed with Error: Object tracking requires license
    found:
      stack: >-
        Error: Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:53628
        objects-start 1.1.1
        Command "objects-start" on "http://127.0.0.1:53628" failed with Error: Object
        tracking requires license
        at ChildProcess.exithandler (child_process.js:203:12)
        at emitTwo (events.js:87:13)
        at ChildProcess.emit (events.js:172:7)
        at maybeClose (internal/child_process.js:817:16)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
      message: >
        Command failed: /bin/sh -c
        /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:53628
        objects-start 1.1.1
       Command "objects-start" on "http://127.0.0.1:53628" failed with Error: Object
        tracking requires license
      killed: false
      code: 1
      signal: null
      cmd: '/bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:53628 objects-start 1.1.1'
    at:
      file: /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js
      line: 25
      column: 9
    source: |
      t.ifError(err);
    stack: |
      /home/test/strongloop_pck/strong-pm/test/test-server-metadata.js:25:9
      ChildProcess.exithandler (child_process.js:210:5)
      maybeClose (internal/child_process.js:817:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

Bail out! # Command failed: /bin/sh -c /home/test/strongloop_pck/strong-pm/bin/sl-pmctl.js -C http://127.0.0.1:53628 objects-start 1.1.1 Commad "objects-start" on "http://127.0.0.1:53628" failed with Error: Object tracking requires license

/**************************/
Yes , i am testing this on a POWER-RHEL 7.1 IBM side vm , as i have been assigned this task . Any pointers how to skip all tests which require licenses ?

**Comment by ghatwala** [comment]
@rmg  -  It appears to me the license based test cases are not getting skipped , please conform at your end  , if license based test cases are being skipping ?

**Comment by ghatwala** [comment]
@rmg -  An update .
I manually added those changes in my local repository , guess your pull request hasn't been merged as yet. By doing that if i manually run test case as - 
[test@pts00449-vm19 strong-pm]$ node test/test-server-metadata.js  -- This test case passes.
But if i run  all combined test cases

[test@pts00449-vm19 strong-pm]$npm test
/*********************/
./test/test-pmctl-ssh2.js working dir for /home/test/strongloop_pck/strong_pm_nolic/strong-pm/test/test-pmctl-ssh2.js is /home/test/strongloop_pck/strong_pm_nolic/strong-pm
exec `git clean -f -d -x .`
exec `git init`
exec `git add .`
exec `git commit --author="sl-pm-test <nobody@strongloop.com>" -m initial`
exec `node "/home/test/strongloop_pck/strong_pm_nolic/strong-pm/node_modules/strong-build/bin/sl-build.js" --install --commit`
test/app built succesfully
pmcli: /home/test/strongloop_pck/strong_pm_nolic/strong-pm/bin/sl-pm.js [ '--base=.strong-pm', '--listen=0' ]
sl-pm.js(13802): StrongLoop PM v5.0.1 (API v6.1.0) on port `38127`
sl-pm.js(13802): Base folder `/tmp/d20151106-13636-1jsugmr0/.strong-pm`
sl-pm.js(13802): Applications on port `3000 + service ID`
Browse your REST API at http://127.0.0.1:38127/explorer
Listening port: 38127
pmurl: http://127.0.0.1:38127
Run request for commit "1/deploy/default/master" on current (none)
Start Runner: commit 1/deploy/default/master
2015-11-06T13:02:16.607Z pid:13870 worker:0 INFO strong-agent v2.0.2 profiling app 'test-app' pid '13870'
2015-11-06T13:02:16.613Z pid:13870 worker:0 INFO strong-agent[13870] started profiling agent
2015-11-06T13:02:16.614Z pid:13870 worker:0 INFO supervisor starting (pid 13870)
2015-11-06T13:02:16.616Z pid:13870 worker:0 INFO strong-agent strong-agent using strong-cluster-control v2.2.1
2015-11-06T13:02:16.618Z pid:13870 worker:0 INFO supervisor reporting metrics to `internal:`
2015-11-06T13:02:16.633Z pid:13870 worker:0 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-11-06T13:02:16.633Z pid:13870 worker:0 Please contact sales@strongloop.com for assistance.
2015-11-06T13:02:16.636Z pid:13870 worker:0 INFO supervisor size set to 1
2015-11-06T13:02:16.823Z pid:13870 worker:0 INFO supervisor started worker 1 (pid 13875)
2015-11-06T13:02:16.824Z pid:13870 worker:0 INFO supervisor resized to 1
2015-11-06T13:02:17.070Z pid:13875 worker:1 INFO strong-agent v2.0.2 profiling app 'test-app' pid '13875'
2015-11-06T13:02:17.084Z pid:13875 worker:1 INFO strong-agent[13875] started profiling agent
2015-11-06T13:02:17.094Z pid:13875 worker:1 PID 13875
2015-11-06T13:02:17.094Z pid:13875 worker:1 CWD /tmp/d20151106-13636-1jsugmr0/.strong-pm/svc/1/work/a6c18e1a53b72c15b72464a9bea46844e96b59bc.1446814933646
2015-11-06T13:02:17.094Z pid:13875 worker:1 ENV.supervisor_profile: undefined
2015-11-06T13:02:17.094Z pid:13875 worker:1 ENV.PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/home/test/strongloop_pck/strong_pm_nolic/strong-pm/node_modules/.bin:/usr/local/rvm/gems/ruby-2.1.2/bin:/usr/local/rvm/gems/ruby-2.1.2@global/bin:/usr/local/rvm/rubies/ruby-2.1.2/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/local/rvm/bin:/home/test/.local/bin:/home/test/bin
2015-11-06T13:02:17.094Z pid:13875 worker:1 ENV.PWD: /tmp/d20151106-13636-1jsugmr0/.strong-pm/svc/1/work/current
2015-11-06T13:02:17.098Z pid:13875 worker:1 INFO strong-agent not profiling, agent metrics requires a valid license.
2015-11-06T13:02:17.098Z pid:13875 worker:1 Please contact sales@strongloop.com for assistance.
2015-11-06T13:02:17.108Z pid:13875 worker:1 pid 13875 listening on 3001
sl-pm.js: Service "1" listening on *:3001
/*******************/

This fails with bail out errors.

**Comment by ghatwala** [comment]
@rmg : i have tried with last merged code with changes to skip the license related test cases , but still above are being noticed . Kindly verify if the license based test cases are being skipped at ur end as well ?

**Comment by rmg** [comment]
@ghatwala do you have `STRONGLOOP_LICENSE` set as an environment variable? It will only skip the tests if you do **not** have that env var set.

**Comment by rmg** [comment]
I was able to get ssh access to a PPCle vm, build node-v4.2.2 from source, clone this git repo, and run `npm install && npm test`. Some regular dev environment setup aside (ssh-agent and docker), it ran without a hitch.

We could make the ssh tests a little more intelligent and add some docker detection to the test-install-systemd tests, but otherwise it was pretty straight forward _for someone familiar with strong-pm_.

I'm going to close this now, since the core of the problem seems to have been resolved: `npm test` of strong-pm does indeed work on Linux on PPCle.
